gcp/src/main/java/org/apache/iceberg/gcp/gcs/OAuth2RefreshCredentialsHandler.java

```diff
@@ -49,6 +53,11 @@ private OAuth2RefreshCredentialsHandler(Map<String, String> properties) {
     Preconditions.checkArgument(
         null != properties.get(GCPProperties.GCS_OAUTH2_REFRESH_CREDENTIALS_ENDPOINT),
         "Invalid credentials endpoint: null");
+    Preconditions.checkArgument(
+        null != properties.get(CatalogProperties.URI), "Invalid catalog endpoint: null");
+    this.credentialsEndpoint =
```

Comment:
Why aren't we doing the same thing here that we do in ADLS?

```java
RESTUtil.resolveEndpoint(
            properties.get(CatalogProperties.URI),
            properties.get(ADLS_REFRESH_CREDENTIALS_ENDPOINT));
```

Reply:
we're doing it for GCP in `GCPProperties`: https://github.com/apache/iceberg/blob/978189cbe33762aa5fe2a76c2028182564d1eec1/gcp/src/main/java/org/apache/iceberg/gcp/GCPProperties.java#L109-L112
For ADLS we're also doing the same in `AzureProperties`: https://github.com/apache/iceberg/blob/68f8053d67aaae7c7cf7b13045e2ed4146fa45f9/azure/src/main/java/org/apache/iceberg/azure/AzureProperties.java#L89-L92
For S3 it happens in `AwsClientProperties`: https://github.com/apache/iceberg/blob/7d0395d4b6f8d690d70910fe2018a459582960e3/aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java#L109-L111